app/javascript/actions.js

```diff
@@ -0,0 +1,46 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+
+export const fetchRandomMessage = createAsyncThunk(
+  "randomMessage/fetchRandomMessage",
+  async () => {
+    const response = await fetch("/api/v1/random/message");
+    if (!response.ok) {
+      throw new Error(`Failed to fetch: ${response.statusText}`);
+    }
+
+    const data = await response.json();
+    return data.message;
+  }
+);
+
+const randomMessageSlice = createSlice({
+  name: "randomMessage",
+  initialState: {
+    message: "",
+    status: "idle",
+    error: null,
+  },
+  reducers: {
+    setRandomMessage: (state, action) => {
+      state.message = action.payload;
+    },
```

Comment:
- [ ] As we are not changing the internal state from any component, the usage of `setRandomMessage` is unnecessary. I recommend removing it.